angular-oauth2-oidc/src/oauth-service.ts

```diff
@@ -1085,8 +1142,36 @@ export class OAuthService
 
         options = options || { };
 
-        let parts: object;
+        if (!this.requestAccessToken && !this.oidc) {
+            return Promise.reject('Either requestAccessToken or oidc or both must be true.');
+        }
+
+        if (!window.location.hash && window.location.search) {
+            return this.tryLoginAuthorizationCode();
+        } else {
+            return this.tryLoginImplicit(options);
+        }
+    };
+
+    private tryLoginAuthorizationCode(): Promise<void> {
 
+        var parameter = window.location.search.split("&")[0].replace("?","").split("=");
+        if (parameter[0] == 'code') {
+            var code = parameter[1];
+
+            return new Promise((resolve, reject) => {
+                this.getTokenFromCode(code).then(result => {        
+                  resolve();
+                });
```

Comment:
I'm getting `Error: Uncaught (in promise): HttpErrorResponse: `, might want to change this to:
```
                this.getTokenFromCode(code).then(result => {
                  resolve();
                }).catch(err => {
                  reject(err);
                })
```
 With the above change im able the catch and handle this correctly.

Reply:
Hi @ceeram 

You have a very good point here. I have tested and added your suggestion to the PR.
I'm not doing much JS in my day job so any Promise help I can get is appreciated. Thanks.

Reply:
Im not very proficient in JS myself, just noticed this was the only way correctly handle a failed request. I have implemented your code and other then the missing reject, it works fine. I had to implement refreshing tokens in my application code though. Hope the PR gets merged an put in a new release soon. Thanks for your work on this.

Reply:
I also need to have tokens refreshed, and I think it should be the responsibility of the library to do it.
I have also been working on a solution for that. Please see my WIP solution at https://github.com/bechhansen/angular-oauth2-oidc/commit/d237f7046f63e21a98e8a69c1661438aa4daed76
If my two existing pull requests get accepted, if will finish this work and do another PR for this improvement.